app/server/ruby/bin/daemon.rb

```diff
@@ -610,13 +629,22 @@ def initialize(ports, kill_switch)
         @tau_pid                       = Promise.new
 
         Util.log "Daemon listening to info from Tau on port #{daemon_port}"
-        @udp_osc_server = SonicPi::OSC::UDPServer.new(daemon_port) do |address, args, sender_addrinfo|
-          Util.log "Daemon received UDP data from Tau: #{[address, args, sender_addrinfo].inspect}"
-        end
+
+        # For debugging purposes:
+        # @udp_osc_server = SonicPi::OSC::UDPServer.new(daemon_port) do |address, args, sender_addrinfo|
+        #   Util.log "Daemon received UDP data from Tau: #{[address, args, sender_addrinfo].inspect}"
+        # end
+
+        @udp_osc_server = SonicPi::OSC::UDPServer.new(daemon_port)
 
         @udp_osc_server.add_method("/tau/pid") do |args|
-          Util.log "Daemon received Pid data from Tau: #{args.inspect}"
-          @tau_pid.deliver!(args[0], false) if args
+          # Util.log "Daemon received Pid data from Tau: #{args.inspect}"
+          # Util.log "token: #{token}"
+          if args[0] && args[0] == token
+            @tau_pid.deliver!(args[1], false) if args[1]
+            # Util.log "Daemon staying alive for longer..."
+            queue << true
```

Comment:
```suggestion
```

Were these two lines unintentionally copied from the keep-alive code? There is no `queue` in this scope and it causes (seemingly harmless) error messages in the console

<details>
<summary>Console error</summary>

```
OSC handler exception for address: /tau/pid
undefined local variable or method `queue' for #<SonicPi::Daemon::TauBooter:0x0000000001755ae8>
["/home/lily/src/sonic-pi/app/server/ruby/bin/daemon.rb:646:in `block in initialize'", "/home/lily/src/sonic-pi/app/server/ruby/lib/sonicpi/osc/udp_server.rb:81:in `handle_data'", "/home/lily/src/sonic-pi/app/server/ruby/lib/sonicpi/osc/udp_server.rb:103:in `block in start_listener'", "/home/lily/src/sonic-pi/app/server/ruby/lib/sonicpi/osc/udp_server.rb:86:in `loop'", "/home/lily/src/sonic-pi/app/server/ruby/lib/sonicpi/osc/udp_server.rb:86:in `start_listener'", "/home/lily/src/sonic-pi/app/server/ruby/lib/sonicpi/osc/udp_server.rb:42:in `block in initialize'"]
```
</details>

Reply:
Excellent catch, thank-you!